be/src/exec/tablet_info.cpp

```diff
@@ -492,6 +492,13 @@ void OlapTablePartitionParam::_compute_hashes(Chunk* chunk, std::vector<uint32_t
         _distributed_columns[i] = chunk->get_column_by_slot_id(_distributed_slot_descs[i]->id()).get();
         _distributed_columns[i]->crc32_hash(&(*indexes)[0], 0, num_rows);
     }
+
+    // if no distributed columns, use random number
+    if (_distributed_slot_descs.size() == 0) {
+        for (auto i = 0; i < num_rows; ++i) {
+            (*indexes)[i] = _rand.Next();
```

Comment:
Did you test its performance?
Why not just choose a first value and just increment it.